Polish and cleanup (#1)
neoforged/Documentation: docs/gettingstarted/index.md

# Getting Started with Forge

:::caution
Please note that this documentation may not be up to date considering the recent creation of NeoForged.

> Until NeoForged releases it's first version, you should refer to Forge documentation. The Forge documentation for 1.20 have been archived here.

Reviewer: ```suggestion
Until NeoForged releases its first version, you should refer to Forge documentation. The Forge documentation for 1.20 has been archived here.
```
